### DISCUSSION TITLE
For view component, is it possible to have a disappear or hide method?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1043] [Ideas]
Hi:

I tried to remove temporary view component in this way but it cause concurrent modification exception
`explo.setOnCycleFinished(()->entity.getViewComponent().removeChild(explo));`
explo is a temporary view component
so is it possible to have a switch that I could tune off rendering of this view component ?
e.g. explo.disappear() or hide();
and explo.appear() or .show();
and of course the view component is rendered by default

For now we set opacity to 0 & 1 for temporary usage
```java
      explo.setOpacity(1);
      explo.setOnCycleFinished(()->explo.setOpacity(0));
      explo.play();
```
Not sure this is a efficient way, I guess the code still will try to render the component...

Thanks

**Comment by AlmasB** [comment]
Hia, if `explo` is a javafx Node, then how about `.setVisible(true / false)`?

**Reply by chengenzhao** [reply]
Wow, didn't thought it was a JavaFX node
Thanks
